tests/unit/core_tests.py

```diff
@@ -152,6 +152,23 @@ def test_context(self):
             self.assertEqual(mock_observer.on_start.call_count, 1)
         self.assertEqual(mock_observer.on_finish.call_count, 1)
 
+    def test_context_with_tags(self):
+        mock_observer = mock.Mock(spec=SpanObserver)
+
+        span = make_test_span()
+        span.register(mock_observer)
+
+        tags = {
+            "k1": "v1",
+            "k2": "v2",
+        }
+        with span.with_tags(tags) as span:
+            self.assertEqual(mock_observer.on_start.call_count, 1)
+            mock_observer.on_set_tag.assert_has_calls(
+                [mock.call(k, v) for k, v in tags.items()], any_order=True
+            )
```

Comment:
🔕 nit: I usually tend towards being explicit with this kind of stuff in tests, accidentally checked that `None == None` one too many times 🤷‍♂️ 😅 

```suggestion
            mock_observer.on_set_tag.assert_has_calls([
                mock.call("k1", "v1"),
                mock.call("k2", "v2"),
            ])
```

Reply:
Updated :-) Thanks for calling me out on my laziness :p